src/main/java/seedu/recipe/model/plan/PlannedRecipe.java

```diff
@@ -1,53 +1,121 @@
 package seedu.recipe.model.plan;
 
+import java.util.HashSet;
+import java.util.List;
 import java.util.Objects;
+import java.util.Set;
 
 import seedu.recipe.model.Date;
 import seedu.recipe.model.recipe.Recipe;
+import seedu.recipe.model.recipe.exceptions.DuplicateRecipeException;
 
 /**
  * Represents a planned recipe in the planned recipes book.
+ * Stores the list of all recipes planned on a date.
  * Guarantees: details are present and not null, field values are validated, immutable.
  */
-public class PlannedRecipe {
+public class PlannedRecipe implements Comparable<PlannedRecipe> {
 
-    private Recipe recipe;
+    private List<Recipe> recipes;
     private Date date;
 
-    public PlannedRecipe(Recipe recipe, Date date) {
-        this.recipe = recipe;
+    public PlannedRecipe(List<Recipe> recipes, Date date) {
+        this.recipes = recipes;
         this.date = date;
     }
 
-    public Recipe getRecipe() {
-        return recipe;
+    /**
+     * Replaces the entire planned recipes list with {@code recipes}.
+     */
+    public void setRecipes(List<Recipe> recipes) {
+        this.recipes = recipes;
     }
 
-    public void setRecipe(Recipe recipe) {
-        this.recipe = recipe;
+    /**
+     * Replaces the {@code target} Recipe with the {@code newRecipe} recipe in the planned recipe's list of recipes.
+     */
+    public void setRecipe(Recipe target, Recipe newRecipe) {
+        int index = recipes.indexOf(target);
+        recipes.set(index, newRecipe);
     }
 
-    public Date getDate() {
-        return date;
+    /**
+     * Adds the recipes in {@code plannedRecipe} to the recipes in this Planned Recipe object and returns a new
+     * Planned Recipe object.
+     */
+    public PlannedRecipe allPlannedRecipesOnOneDay(PlannedRecipe plannedRecipe) {
+        if (recipes.contains(plannedRecipe)) {
+            throw new DuplicateRecipeException();
+        }
+        List<Recipe> newRecipes = plannedRecipe.getRecipes();
+        newRecipes.addAll(recipes);
+        return new PlannedRecipe(newRecipes, date);
+    }
+
+    /**
+     * Returns true if {@code plannedRecipe} is planned on the same day and all its recipes
+     * can be found in the current PlannedRecipe's list, or if duplicate recipes are found in {@code plannedRecipe}.
+     */
+    public boolean hasSameRecipeInPlan(PlannedRecipe plannedRecipe) {
+        if (plannedRecipe.recipesAreUnique()) {
+            return date.equals(plannedRecipe.date) && recipes.containsAll(plannedRecipe.getRecipes());
```

Comment:
May i clarify if this means that we can only plan one recipe for a day? So we cant have the same meal for lunch and dinner? If so maybe a suggestion might be to compare by timeframe or perhaps have a enum class with meals: breakfast, lunch, dinner, tea or supper! But other than this everything looks good to me! thanks for the hardwork!!

Reply:
Hey! Yes, currently you can only plan only one copy of the same recipe on a day.
Lets discuss this tomorrow? I have some concerns regarding this!